src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/Editor_OutOfProc.cs

```diff
@@ -145,6 +149,26 @@ public void PressDialogButton(string dialogAutomationName, string buttonAutomati
         public void DialogSendKeys(string dialogAutomationName, string keys)
             => _inProc.DialogSendKeys(dialogAutomationName, keys);
 
+        public void FormatDocument() {
+            VisualStudioInstance.VisualStudioWorkspace.WaitForAsyncOperations(FeatureAttribute.Workspace);
+            SendKeys(new KeyPress(VirtualKey.K, ShiftState.Ctrl), new KeyPress(VirtualKey.D, ShiftState.Ctrl));
+        }
+
+        public void FormatSelection() {
+            VisualStudioInstance.VisualStudioWorkspace.WaitForAsyncOperations(FeatureAttribute.Workspace);
+            SendKeys(new KeyPress(VirtualKey.K, ShiftState.Ctrl), new KeyPress(VirtualKey.F, ShiftState.Ctrl));
+        }
+
+        public void Paste(string text)
+        {
+            var thread = new Thread(() => Clipboard.SetText(text));
```

Comment:
Yeah, I know. The only two options I knew of were to do this or go in-proc to get on the UI thread. If I'm missing a better alternative, let me know. I prefer to stay out-of-proc whenever possible, but I'm happy to switch if people are super anti-this.